### PR TITLE
feat: Animated a Progress Bar Component

### PR DESCRIPTION
**Pull request by Mwimwii**
##  Animated a Progress Bar Component #19 

- Yes, an animated progress bar using the `Transition` and `Progress` components from mantine
- This was just an idea I wanted to replicate to break a bit more into `mantine/core` without using `gsap`
- There were some test I had missed for the `formatDay` from the previous PR that I included

### Preview

<a href="https://gyazo.com/9f3d1b79668cf35915df4636916c6c3d">Image from Gyazo</a>

### Changes

- created a  `ScheduleCardProgress` component to perform the onMount animation 
- added `timeElapsedPercent` function to calculate the  time elapsed percentage
- added tests for `formatDay` and `timeElapsedPercent` functions in `utils.test.ts`

### CODE REVIEW

File: src/util.test.ts [hunk 1]
```diff
@@ -41,6 +49,45 @@ describe('Utility Functions', () => {
     });
   });
 
+  describe('timeElapsedPercent', () => {
```
Comment (OlivierJM) [inline src/util.test.ts:52]: 🙏

File: src/components/Schedule/ScheduleCardProgress.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { Progress, Transition } from '@mantine/core';
+
+const ScheduleCardProgess = ({ value }: { value: number }) => {
```
Comment (OlivierJM) [inline src/components/Schedule/ScheduleCardProgress.tsx:4]: Can we also add some tests for this component.

Comment (Mwimwii) [reply src/components/Schedule/ScheduleCardProgress.tsx:4]: I'll try to create a test but I'm not sure how to test transitions. I would like some help on this to be honestly

Comment (OlivierJM) [reply src/components/Schedule/ScheduleCardProgress.tsx:4]: Let's just add some data-testid attribute to the component and we test if those can be found in the document. we can also go a step further by mocking the timeout and verify if progress value has changed.

Comment (Mwimwii) [reply src/components/Schedule/ScheduleCardProgress.tsx]: Thanks, I actually found used found that the `Progress` component has a role which can utilize the `getByRole('progressbar)` function. 

I also made sure to await the `useEffect` with a timeout of 300 to trigger the mounting of the `Progress` Component

Comment (OlivierJM) [reply src/components/Schedule/ScheduleCardProgress.tsx]: That smart 🚀

**Pull request merged**